### PR TITLE
{Packaging} Drop support for Python 2.7

### PR DESCRIPTION
**Pull request by rodrigc**
- Fix https://github.com/Azure/azure-cli/issues/11362 
- Fix https://github.com/Azure/azure-cli/issues/11206

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: tools/setup.py [hunk 1]
```diff
@@ -36,6 +35,10 @@
     'colorama>=0.3.7'
 ]
 
+ _python_version = getattr(sys, "version_info", (0,))
```
Comment (fengzhou-msft) [inline tools/setup.py]: wrong indentation here.

Comment (rodrigc) [reply tools/setup.py]: Fixed

File: tools/setup.py [hunk 2]
```diff
@@ -37,6 +36,10 @@
     'colorama>=0.3.7'
 ]
 
+_python_version = getattr(sys, "version_info", (0,))
```
Comment (haroldrandom) [inline tools/setup.py]: Why use `getattr` instead of call `sys.version_info` directly?

Comment (rodrigc) [reply tools/setup.py]: @haroldrandom Fixed

**Comment by fengzhou-msft** [comment]
Move to S165, plan to upgrade to version 2.1.0 for S165 with the change to drop support for Python 2. Will merge this PR in early February.

**Comment by fengzhou-msft** [comment]
@rodrigc can you merge the latest dev branch and help resolve the conflicts? we're ready to merge this PR.

**Pull request merged**